docs/language/ql-handbook/language.rst

```diff
@@ -162,11 +235,9 @@ For selection identifiers (``a::b``):
 
 For qualified identifiers (``a.b``):
 
--  Define the *current file* as the file the import directive occurs in.
-
--  Determine the current file's *query directory*, if any. Starting with the directory containing the current file, and walking up the directory structure, each directory is checked for a file called ``queries.xml``, containing a single top-level tag named ``queries``, which has a ``language`` attribute set to the identifier of the active database scheme (for example, ``<queries language="java"/>``). The closest enclosing directory is taken as the current file's query directory.
-
--  Build up a list of *candidate search paths*, consisting of the current file's directory, the current file's query directory (if one was determined in the previous step), and the list of directories making up the library path (in order).
+-  Build up a list of *candidate search paths*, consisting of the
+   current file's directory, and each of the directories on the
```

Comment:
The order is actually the "current file's directory", "current file's query directory", then the library path. See `lookup` in `QueryImportResolver` for the code. We also don't seem to mention underscore to space replacement. We try to find the file once without replace underscores with spaces in all section and then with replacement.

Reply:
I couldn't find any code that would look up "current file's query directory" -- where should I look?

Reply:
`QueryImportResolver:58`

Reply:
We actually look up the current folders query directory, but that gets equivalent results.

Reply:
Hmm, I see.

That code looks somewhat iffy, though. It doesn't recognize `qlpack.yml` even if we're `codeql`, and it doesn't use the virtual file system if we're the IDE.

I think we should consider removing that behavior rather than documenting it. The query directory for a `.qll` file will generally already be on the library path (so files relative to there will still be found, although later in the search sequence). And it would make sense to me simply to prepend the query directory for a `.ql` file to the library path once and for all -- that's what `codeql` already does.

Reply:
`QueryDirectory` does now recognise `qlpack.yml` as a query directory. We don't use the virtual file system for anything except ql file contents currently so this matches everything else. This behaviour also matches better what we plan to do with encapsulated packs where unspecified imports will only resolve within a single pack unless we are using backwards compatibility features like libarypaths.

Reply:
OK then. Changed.

The underscore replacement thing seems orthogonal and should go in a separate PR, I think.